rust-runtime/aws-smithy-http-server/src/routing/routers/aws_json.rs

```diff
@@ -0,0 +1,128 @@
+use std::convert::Infallible;
+
+use http::header::ToStrError;
+use tower::{Layer, Service};
+
+use crate::{
+    body::BoxBody,
+    extension::RuntimeErrorExtension,
+    protocols::{AwsJson10, AwsJson11},
+    response::IntoResponse,
+    routing::{tiny_map::TinyMap, Route},
+};
+
+use super::Router;
+
+pub enum Error {
+    NotRootUrl,
+    MethodDisallowed,
+    MissingHeader,
+    InvalidHeader(ToStrError),
+    NotFound,
+}
+
+impl IntoResponse<AwsJson10> for Error {
+    fn into_response(self) -> http::Response<BoxBody> {
+        match self {
+            Error::MethodDisallowed => super::method_disallowed(),
+            _ => http::Response::builder()
+                .status(http::StatusCode::NOT_FOUND)
+                .header("Content-Type", "application/x-amz-json-1.0")
+                .extension(RuntimeErrorExtension::new(
+                    super::UNKNOWN_OPERATION_EXCEPTION.to_string(),
+                ))
+                .body(crate::body::to_boxed(""))
+                .expect("invalid HTTP response"),
+        }
+    }
+}
+
+impl IntoResponse<AwsJson11> for Error {
+    fn into_response(self) -> http::Response<BoxBody> {
+        match self {
+            Error::MethodDisallowed => super::method_disallowed(),
+            _ => http::Response::builder()
+                .status(http::StatusCode::NOT_FOUND)
```

Comment:
Would be nice to debug log the concrete error?

Reply:
Shall we implement `Error` on the router errors and write a general `error!` in the `RouterService::call`?

Reply:
Not `error!`, use debug. Services receive invalid requests from misbehaving/scanning clients all the time, and teams have alarms triggering on logs at the error level.